src/components/App.tsx

```diff
@@ -1,15 +1,44 @@
+import { useState } from 'react'
+import { Modal } from './modal'
+
+const Body = ({setShowModal} : any) => {
+  return (
+    <>
+      <p>
+        In order to close the modal, you can click outside of it. Also, you can click the button
+        on the top right corner or click this button below
+      </p>
+      <button onClick={() => setShowModal(false)}>Click me!</button>
+    </>
+  )
+}
 
 const App = () => {
+  const [showModal, setShowModal] = useState(false)
+
+  const handleShowModal = () => {
+    setShowModal(true)
+  }
+
   return (
     <>
       <h1>Trabe interview</h1>
       <h2>Exercise 2: Modal</h2>
 
       <hr />
 
-    </>
+      <button onClick={handleShowModal}>Show modal</button>
 
-  );
+      <Modal 
+        showModal={showModal} 
+        setShowModal={setShowModal}
+        title='Hello Trabe'
+        Body={() => (<Body setShowModal={setShowModal} />)}
```

Comment:
This is not exactly what we expected as the API to allow closing the modal from the content. The approach is similar. Can you give it another try?.

Reply:
When I look into my code again, I also thought it was a little bit weird. Maybe this is what you were expecting.

<img width="792" alt="Captura de pantalla 2022-03-21 a las 13 07 18" src="https://user-images.githubusercontent.com/51116828/159249171-11af886f-03b7-4b91-8ac3-2c8b719af188.png">